src/Core/Settings.cpp

```diff
@@ -3652,7 +3652,7 @@ Normalize function names to their canonical names
     DECLARE(Bool, enable_early_constant_folding, true, R"(
 Enable query optimization where we analyze function and subqueries results and rewrite query if there are constants there
 )", 0) \
-    DECLARE(Bool, deduplicate_blocks_in_dependent_materialized_views, false, R"(
+    DECLARE(Bool, deduplicate_blocks_in_dependent_materialized_views, true, R"(
```

Comment:
We unfortunately unable to make it by default.
It does not work with async inserts. Im working on it right now. When it is done we will enable it by default.